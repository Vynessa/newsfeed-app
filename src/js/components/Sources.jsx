import React from 'react';
import { Col, Card, Row, Input } from 'react-materialize';
import css from '../../../public/css/main.scss';
import NewsActions from '../actions/NewsActions';
import SourcesStore from '../stores/SourcesStore';

/**
 * @class
 * @param
 */
export default class Sources extends React.Component {
  /**
   *
   */
  constructor(props) {
    super(props);
    this.state = {
      sources: [],
      search: ''
    };
    this.onChange = this.onChange.bind(this);
  }
  /**
   *
   */
  componentWillMount() {
    NewsActions.allSources();
    SourcesStore.addChangeListener(this.onChange);
  }

  componentWillUnmount() {
    SourcesStore.removeChangeListener(this.onChange);
  }

  onChange() {
    this.setState({
      sources: SourcesStore.getAll()
    });
  }

  updateSearch(e) {
    this.setState({
      search: e.target.value.substr(0, 30)
    });
  }
  /**
   *
   */
  render() {
    const filteredSearch = this.state.sources.filter((source) => {
      return source.name.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1;
    });

    const sources = filteredSearch.map((source, index) => {
      // console.log(Object.keys(sources.length));
      return (
        <div className='row'>
          <Col key={`Col-${index}`} m={4} s={12}>
            <Card key={`Card-${index}`} className='teal darken-1' textClassName='white-text' title={source.name} actions={[<a href='#'>View headlines</a>]}>{source.description}
            </Card>
          </Col>
        </div>
      );
    });

    return (
      <div>
        <Row>
          <Input s={3} label="Search Sources"
                  value={this.state.search}
                  onChange={this.updateSearch.bind(this)}/>
        </Row>
        <div className="source">{sources}</div>
      </div>
    );
  }
}
