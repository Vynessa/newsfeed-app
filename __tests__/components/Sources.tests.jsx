import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Sources from '../../src/js/components/Sources.jsx';
import mockedSources from '../../__mocks__/mockSources.json';

describe('Sources Component test suite', () => {
  describe('When the `Sources` component renders', () => {
    // const items = ['Learn react', 'rest', 'go out'];

    it('should match its empty snapshot', () => {
      const tree = renderer.create(
        <Sources sources={[]}/>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  it('should add an item based on the value in the state', () => {
    const component = shallow(<Sources />);
    // const preventDefault = jest.fn();
    component.setState({
      mockedSources
    });
    component.find('div').simulate('card');
    expect(toJson(component)).toMatchSnapshot();
    // expect(preventDefault).toBeCalled();
  });

  it('should pass a selected value to the onChange function', () => {
    const component = shallow(<Sources sortCategory={this.sortCategory}/>);
    component.find('input').simulate('change', { target: {
      value: 'Change function' }
    });

    expect(toJson(component)).toMatchSnapshot();
  });

  // describe('Given the Sources Component test suite', () => {
  //   describe('When the `Sources` component is loaded', () => {
  //     it('Then renders Sources', () => {
  //       const div = document.createElement('app');
  //       ReactDOM.render(<Sources/>, div);
  //     });
  //   });
  // });
});

    // describe('When the `Sources` component renders', () => {
    //   // const items = ['Learn react', 'rest', 'go out'];

    //   it('should match its empty snapshot', () => {
    //     const tree = renderer.create(
    //       <Sources />
    //     ).toJSON();

    //     expect(tree).toMatchSnapshot();
    //   });
    // });

    // describe('When the sources state has data', () => {
    //   it('Then ', () => {
    //     const
    //   });
    // });
