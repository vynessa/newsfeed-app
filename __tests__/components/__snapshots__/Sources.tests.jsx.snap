// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sources Component Test Suite When the \`Sources\` component renders should contain all sources based on the value in the state 1`] = `
<div
  className="sources"
>
  <SourcesList
    categories={Array []}
    search=""
    sortCategory={[Function]}
    sources={[TypeError: Cannot read property 'sources' of undefined]}
    storeItems={[Function]}
    updateSearch={[Function]}
  />
</div>
`;

exports[`Sources Component Test Suite When the \`Sources\` component renders should match its empty snapshot 1`] = `
<div
  className="sources"
>
  <div>
    <div
      className="center-align"
    >
      <div
        className="row"
      >
        <div
          className="col input-field s3"
        >
          <input
            className="search-content"
            defaultValue={undefined}
            id="input_0"
            onChange={[Function]}
            placeholder={undefined}
            type="text"
            value=""
          />
          <label
            className=""
            data-error={undefined}
            data-success={undefined}
            htmlFor="input_0"
          >
            Search Sources
          </label>
        </div>
        <div
          className="col input-field s6 m3"
        >
          <label
            className="active"
            data-error={undefined}
            data-success={undefined}
            htmlFor="input_1"
          >
            Categories:
          </label>
          <select
            className=""
            defaultValue=""
            id="input_1"
            multiple={undefined}
            onChange={[Function]}
          >
            <option
              value=""
            >
              All News Sources
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="center-align"
      >
        <div
          className="preloader-wrapper big active"
        >
          <div
            className="spinner-layer spinner-brown-only"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="circle-clipper right"
            >
              <div
                className="circle"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="clearfix"
    />
  </div>
</div>
`;
